meta-mbl-apps/recipes-support/swupdate/swupdate-headers_2019.11.bb

```diff
@@ -0,0 +1,20 @@
+# Copyright (c) 2019 Arm Limited and Contributors. All rights reserved.
+#
+# SPDX-License-Identifier: MIT
+
+require recipes-support/swupdate/swupdate_${PV}.bb
+require swupdate-append-common.inc
+
+SUMMARY = "Header files for building handlers for swupdate"
+DESCRIPTION = "${SUMMARY}"
+FILESEXTRAPATHS_prepend = "${OEROOT}/layers/meta-swupdate/recipes-support/swupdate/swupdate:"
+
+PACKAGES = "${PN}"
+FILES_${PN} = "${includedir}/*"
```

Comment:
The header files should be included in the `-dev` package, not in the main one.

Reply:
I've been debating something like that with myself for a few days...

If I didn't know anything about the recipes, I'd expect to find headers in "swupdate-dev", but  we don't want to use that name to avoid clashes with the swupdate recipe. In my mind the name "swupdate-headers" means "unofficial-swupdate-dev", or "not-quite-swupdate-dev"... I.e. it's sort-of an unofficial part of the swupdate recipe rather than a completely new thing that should have it's own dev package ("swupdate-headers-dev").

I'm not convinced either way though.

Reply:
By default bitbake put the headers in the `-dev` package and when you have `DEPENDS` on that package, bitbake install the `-dev` package in the `recipe-sysroot` directory. This is the way it works.

By adding headers in the main package we are deviating from the standard. If we really follow this path we should at least add a comment in the recipe explaning the reason for not following the standard.

Reply:
> we should at least add a comment in the recipe explaning the reason for not following the standard.

Good point. There are a lot of non-obvious things going on in the recipes we've added and modified here. I'll add some comments.

Reply:
I've added some comments in https://github.com/ARMmbed/meta-mbl/pull/811/commits/e328c7cde35a9fdd84e61ef81e7db49508a3490e.

Reply:
Thanks for adding the comments, but you still haven't convinced me why we are not following the yocto standard and installing headers in the `swupdate-headers` package instead of `swupdate-headers-dev`.

For instance, on the oe-core we have the `swupdate-headers` and the headers are installed in the `swupdate-headers-dev` package.

Reply:
> For instance, on the oe-core we have the swupdate-headers and the headers are installed in the swupdate-headers-dev package.

I don't know what you mean about oe-core... swupdate isn't in oe-core.

Anyway, https://github.com/ARMmbed/meta-mbl/pull/811/commits/e85d5b822ac6811c23929b349f11bb972f2a835a.

Reply:
Sorry. There was a typo in my comment. For the oe-core I meant `linux-libc-headers` not `swupdate-headers`.